erpnext/selling/doctype/sales_order/sales_order.py

```diff
@@ -114,6 +116,15 @@ def validate_delivery_date(self):
 
 		self.validate_sales_mntc_quotation()
 
+	def validate_delivery_day(self):
+		delivery_days = frappe.db.get_value("Customer", self.customer, "delivery_days")
+		if delivery_days:
+			day = calendar.day_name[getdate(self.delivery_date).weekday()]
+			weekdays = json.loads(delivery_days)
+			if not day in weekdays:
+				frappe.msgprint(_("This order is set to be delivered on a '{0}', but {1} only accepts deliveries on {2}").
+							format(day, self.customer, ', '.join(weekdays)))
```

Comment:
Use the `comma_and` function to format this nicely

```suggestion
							format(day, self.customer, comma_and(weekdays)))
```